merlin/models/tf/models/base.py

```diff
@@ -1154,6 +1310,8 @@ def fit(
         if pre:
             self._reset_compile_cache()
             self.train_pre = pre
+            if isinstance(self.train_pre, SequenceTransform):
+                self.train_pre.on_train_begin()
```

Comment:
I think that using methods named like callback (`on_train_begin()` and `on_test_begin()`) might confuse the user, as it might think that a callback event is being called here. And they might think that such methods for `train_end` or `test_end` might also be available...
I would suggest renaming these methods to something like `configure_for_train()`,  `configure_for_test()`